build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -e
+
+eval "$(pyenv init --path)"
+eval "$(pyenv virtualenv-init -)"
+UNAME_S=$(uname -s)
+UNAME_M=$(uname -m)
+
+if [ -z "$PYENV_VERSION" ]; then
+    echo "Warning: PYENV_VERSION is not set. Defaulting to Python 3.11."
+    PYENV_VERSION="3.11"
+fi
+
+# Need to tweak pyenv if darwin/amd64 due to Dockerfile pyenv not building with --enable-shared
```

Comment:
For reasons currently unknown to me, only on darwin/amd64, Pyinstaller runs into an issue where it needs to find libpython as a shared library. This logic reinstalls pyenv python3.11 with dynamic linking so Pyinstaller is appeased

Reply:
very good catch -- I didn't know about this and it's an important bug

more details: on x86 machines, libpython is a `.a` instead of a `.dylib`. More details in [APP-4596](https://viam.atlassian.net/browse/APP-4596) if you want

this is a vendor thing -- our options here are:
1. email them and ask to fix (already sent)
2. build python ourselves somewhere in the automatic setup parts
3. discontinue x86 mac support (will get less + less common)

for now focused on option 1